htdocs/core/lib/functions.lib.php

```diff
@@ -8876,21 +8876,22 @@ function dol_osencode($str)
  * 		@param	string	$fieldid		Field to get
  *      @param  int		$entityfilter	Filter by entity
  *      @param	string	$filters		Filters to add. WARNING: string must be escaped for SQL and not coming from user input.
+ *      @param	bool    $useCache       If true (default), cache will be queried and updated.
  *      @return int						<0 if KO, Id of code if OK
  *      @see $langs->getLabelFromKey
  */
-function dol_getIdFromCode($db, $key, $tablename, $fieldkey = 'code', $fieldid = 'id', $entityfilter = 0, $filters = '')
+function dol_getIdFromCode($db, $key, $tablename, $fieldkey = 'code', $fieldid = 'id', $entityfilter = 0, $filters = '', $useCache = true)
 {
-	global $cache_codes;
+	static $cache_codes = array();
```

Comment:
Why using static instead of global ?

Reply:
It avoids overcrowding the global namespace and ensures the variable stays private (can't be read or written from the outside, which can be good or bad depending on the situation). I saw that in v21, this variable is replaced with `$conf->cache`, which I assume is used by other functions, so in this regard, a global variable makes more sense.
Feel free to revert this line.